fluent/fluent.go

```diff
@@ -194,7 +205,11 @@ func (f *Fluent) close() (err error) {
 
 // connect establishes a new connection using the specified transport.
 func (f *Fluent) connect() (err error) {
-	f.conn, err = net.DialTimeout("tcp", f.Config.FluentHost+":"+strconv.Itoa(f.Config.FluentPort), f.Config.Timeout)
+	if strings.HasPrefix(f.Config.FluentNetwork, "unix") {
```

Comment:
`unixdomainsocket` like string will be matched here.
Is this intended behavior?